packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -39,10 +55,35 @@ export class CoinbaseWalletProvider extends ProviderEventEmitter implements Prov
     const signerType = loadSignerType();
     if (signerType) {
       this.signer = this.initSigner(signerType);
+      logSignerLoadedFromStorage({ signerType });
     }
   }
 
   public async request<T>(args: RequestArguments): Promise<T> {
+    // correlation id across the entire request lifecycle
+    const correlationId = crypto.randomUUID();
```

Comment:
not sure if this solves @cb-jake's comment about using the crypto package directly, just flagging

Reply:
this will work on the web

https://developer.mozilla.org/en-US/docs/Web/API/Crypto/subtle